spec/controllers/admin/settings_controller_spec.rb

```diff
@@ -0,0 +1,181 @@
+          expect(flash[:error]).to(
+            include("You are not permitted to change the following settings: Suspend filter counts, Enable test caching, Cache expiration, Tag wrangling off, Downloads enabled")
+          )
+        end
+
+        it "allows admins with policy_and_abuse role to update spam setting" do
+          admin.update(roles: ['policy_and_abuse'])
+          fake_login_admin(admin)
+
+          put :update, params: { id: setting.id, admin_setting: { hide_spam: "1" } }
+
+          it_redirects_to_with_notice(admin_settings_path, "Archive settings were successfully updated.")
+        end
+      end
+
+      context 'when admin has support role' do
+        it "prohibits admins with support role to update forbidden settings" do
+          admin.update(roles: ['support'])
+          fake_login_admin(admin)
+
+          put :update, params: {
+            id: setting.id,
+            admin_setting: {
+              disable_support_form: "1",
+              disabled_support_form_text: "Disable support",
+              tag_wrangling_off: "0",
+              downloads_enabled: "1",
+              enable_test_caching: "0",
+              hide_spam: "1"
+            } 
+          }
+
+          expect(response).to render_template('index')
+          expect(flash[:error]).to(
+            include("You are not permitted to change the following settings: Enable test caching, Tag wrangling off, Downloads enabled, Hide spam")
+          )
+        end
+
+        it "allows admins with support role to update spam setting" do
+          admin.update(roles: ['support'])
+          fake_login_admin(admin)
+
+          put :update, params: {
+            id: setting.id,
+            admin_setting: {
+              disable_support_form: "1",
+              disabled_support_form_text: "Disable support"
+            }
+          }
+
+          it_redirects_to_with_notice(admin_settings_path, "Archive settings were successfully updated.")
+        end
+      end
+
+      context 'when admin has tag_wrangling role' do
+        it "prohibits admins with tag_wrangling role to update forbidden settings" do
+          admin.update(roles: ['tag_wrangling'])
+          fake_login_admin(admin)
+
+          put :update, params: {
+            id: setting.id,
+            admin_setting: {
+              disable_support_form: "1",
+              disabled_support_form_text: "Disable support",
+              tag_wrangling_off: "0",
+              downloads_enabled: "1",
+              enable_test_caching: "0",
+              hide_spam: "1"
+            } 
+          }
+
+          expect(response).to render_template('index')
+          expect(flash[:error]).to(
+            include("You are not permitted to change the following settings: Enable test caching, Downloads enabled, Hide spam, Disable support form, Disabled support form text")
+          )
+        end
+
+        it "allows admins with policy_and_abuse role to update spam setting" do
```

Comment:
It looks like this is actually testing `it "allows admins with tag_wrangling role to turn wrangling off"`